src/System.IO.FileSystem.Watcher/tests/FileSystemWatcher.MultipleWatchers.cs

```diff
@@ -0,0 +1,248 @@
+                watcher2.EnableRaisingEvents = true;
+                watcher3.EnableRaisingEvents = true;
+
+                File.Delete(fileName);
+                Assert.True(WaitHandle.WaitAll(new[] { autoResetEvent1, autoResetEvent2, autoResetEvent3 }, WaitForExpectedEventTimeout_NoRetry));
+                
+                File.Create(fileName).Dispose();
+                watcher1.EnableRaisingEvents = false;
+
+                File.Delete(fileName);
+                Assert.False(autoResetEvent1.WaitOne(WaitForExpectedEventTimeout_NoRetry));
+                Assert.True(WaitHandle.WaitAll(new[] { autoResetEvent2, autoResetEvent3 }, WaitForExpectedEventTimeout_NoRetry));
+            }
+        }
+
+        [OuterLoop]
+        [Fact]
+        [PlatformSpecific(TestPlatforms.OSX)]
+        public void FileSystemWatcher_File_Rename_MultipleWatchers_NotAffectEachOther()
+        {
+            using (var testDirectory = new TempDirectory(GetTestFilePath()))
+            using (var file = new TempFile(Path.Combine(testDirectory.Path, "file")))
+            using (var watcher1 = new FileSystemWatcher(testDirectory.Path, Path.GetFileName(file.Path)))
+            using (var watcher2 = new FileSystemWatcher(testDirectory.Path, Path.GetFileName(file.Path)))
+            {
+                AutoResetEvent autoResetEvent1_created = WatchCreated(watcher1, new[] { Path.Combine(testDirectory.Path, "file") }).EventOccured;
+                AutoResetEvent autoResetEvent1_deleted = WatchDeleted(watcher1, new[] { Path.Combine(testDirectory.Path, "file") }).EventOccured;
+                AutoResetEvent autoResetEvent2_created = WatchCreated(watcher2, new[] { Path.Combine(testDirectory.Path, "file") }).EventOccured;
+                AutoResetEvent autoResetEvent2_deleted = WatchDeleted(watcher2, new[] { Path.Combine(testDirectory.Path, "file") }).EventOccured;
+
+                watcher1.EnableRaisingEvents = true;
+                watcher2.EnableRaisingEvents = true;
+
+                string filePath = file.Path;
+                string filePathRenamed = file.Path + "_renamed";
+
+                File.Move(filePath, filePathRenamed);
+                Assert.True(WaitHandle.WaitAll(
+                	new[] { autoResetEvent1_created, autoResetEvent1_deleted, autoResetEvent2_created, autoResetEvent2_deleted }, 
+                	WaitForExpectedEventTimeout_NoRetry));
+
+                File.Move(filePathRenamed, filePath);
+                watcher1.EnableRaisingEvents = false;
+
+                File.Move(filePath, filePathRenamed);
+                Assert.False(WaitHandle.WaitAll(
+                	new[] { autoResetEvent1_created, autoResetEvent1_deleted }, 
+                	WaitForExpectedEventTimeout_NoRetry));
+                Assert.True(WaitHandle.WaitAll(
+                	new[] { autoResetEvent2_created, autoResetEvent2_deleted }, 
+                	WaitForExpectedEventTimeout_NoRetry));
+            }
+        }
+    }
+}
```

Comment:
Maybe also a test that enables and disabled the same watcher a bunch of times in a loop, and then enables it again, and validates that an expected event is raised once and only once?

And one for two watchers watching two different paths, that makes sure they get the right event raised and not the other's?

Reply:
Added `FileSystemWatcher_File_Create_WatchOwnPath` test